### PR TITLE
Feature/correlation rust

### PR DESCRIPTION
**Pull request by Alexanderlacuna**
This PR contains code for calling rust correlation code  in this repo:
https://github.com/Alexanderlacuna/correlation_rust

### CODE REVIEW

**Comment by Alexanderlacuna** [comment]
@BonfaceKilz  this is ready for merge

File: gn3/computations/rust_correlation.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import os
+
+
+from gn3.computations.qtlreaper import create_output_directory
+from gn3.random import random_string
+from gn3.settings import CORRELATION_COMMAND
+from gn3.settings import TMPDIR
+
+
+def generate_input_files(dataset: list[str], output_dir: str = TMPDIR):
```
Comment (BonfaceKilz) [inline gn3/computations/rust_correlation.py]: *suggestion: nitpick:* Let's try to be consistent with our naming. I see we have 2 different patterns: "generate_*" vs "create_*". We should choose one and stick to it.

Comment (BonfaceKilz) [inline gn3/computations/rust_correlation.py]: *issue:* You should also specify the return type of this.

File: gn3/computations/rust_correlation.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    tmp_file = os.path.join(tmp_dir, f"{random_string(10)}.txt")
+
+    with open(tmp_file, "w", encoding="utf-8") as file_writer:
+
+        file_writer.write("\n".join(dataset))
+    return (tmp_dir, tmp_file)
+
+
+def generate_json_file(**kwargs):
```
Comment (BonfaceKilz) [inline gn3/computations/rust_correlation.py]: **issue:** preferably make the args explicit. Do you see them varying?  Something like:
```suggestion
def generate_json_file(dir: str, file_name:str, method: str, delimiter: str) -> str:
```

File: gn3/computations/rust_correlation.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    json_file = generate_json_file(**
+                                   {"tmp_dir": tmp_dir, "tmp_file": tmp_file,
+                                    "method": method, "delimiter": delimiter,
+                                    "x_vals": trait_vals})
+
+    command_list = [CORRELATION_COMMAND, json_file, TMPDIR]
+
+    results = subprocess.run(command_list, check=True)
+
+    return results
```
Comment (BonfaceKilz) [inline gn3/computations/rust_correlation.py]: **tip:** If and when I can, I avoid naming variables.

```suggestion
    return subprocess.run(command_list, check=True)
```

File: tests/unit/computations/test_rust_correlation.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+from gn3.computations.rust_correlation import generate_json_file
+from gn3.computations.rust_correlation import generate_input_files
+from gn3.computations.rust_correlation import parse_correlation_output
+
+
+@pytest.mark.unit_test
+def test_run_correlation():
+    """test calling rusts' correlation """
+
```
Comment (BonfaceKilz) [inline tests/unit/computations/test_rust_correlation.py]: **issue:** This test does nothing.  Remove it.

**Review by BonfaceKilz — CHANGES_REQUESTED** [review]
@Alexanderlacuna I've left some comments you can work on.  Please address them so that we can hopefully merge by the end of today. Finally, thanks for your contribution.

File: gn3/computations/rust_correlation.py (hunk 2 above)
Comment (Alexanderlacuna) [reply gn3/computations/rust_correlation.py]: they are constants

File: gn3/computations/rust_correlation.py [hunk 5]
```diff
@@ -30,18 +31,16 @@ def generate_input_files(dataset: list[str], output_dir: str = TMPDIR):
     return (tmp_dir, tmp_file)
 
 
-def generate_json_file(**kwargs):
+def generate_json_file(tmp_dir, tmp_file, method, delimiter, x_vals) -> str:
```
Comment (BonfaceKilz) [inline gn3/computations/rust_correlation.py:34]: *nitpick:* @Alexanderlacuna Could you type hint this?

**Review by BonfaceKilz — APPROVED** [review]
@Alexanderlacuna This LGTM.  Next step, send me a patch with the correlations work either on mail or on matricx; and then I can merge this.

**Comment by BonfaceKilz** [comment]
@Alexanderlacuna Also, @Alexanderlacuna make sure all the linters work.  I don't want regressions on our CI...

**Comment by BonfaceKilz** [comment]
Alexander Kabui ***@***.***> writes:

> @BonfaceKilz  this is ready for merge 
>
Cool. Let me have a look.

**Pull request merged**